lib/cn/kblock_builder.c

```diff
@@ -138,10 +138,10 @@ struct curr_kblock {
     u32 num_keys;
     u32 num_tombstones;
 
-    uint max_size;
-    uint max_pgc;
-    uint blm_pgc;
-    uint wbt_pgc;
+    uint64_t max_size;
+    uint64_t max_pgc;
+    uint     blm_pgc;
+    uint     wbt_pgc;
```

Comment:
You've switched a subset of these scalars from `uint` to `uint64_t`.  I would switch these back to `uint` or move them all to `uint64_t`.

Reply:
Well I didn't want to dig too deeply here. kblock_builder is at risk for a lot of integer overflow in my opinion. I only changed what had a dependency on mblocksz although I think these should actually be uint32_t

Reply:
pgc means page count, so (n * 4096). 256 pages per megabyte, so with kblocks being quite small (32M max???) they should never be in danger of overflowing.  dunno about the others...   does this patch change the max kblock size?